Flask server

dlahtine opened the pull request:
Contains the server portion.

To test:

1. `parcel build web/index.html` from root
2. `cmake --build build --target all` from root (may have to create build folder)
3. use python to run FlaskServer.py
4. In browser, open http://loaclhost:5000/web/index.html to view simulation
5. Download one of the test bsp files from google drive, and then visit http://loaclhost:5000/data.trajectory to upload it to the server.

You should get some debug messages, but no complaints (other than maybe a missing icon file)

dlahtine committed:
created basic flask server

dlahtine committed:
Moved flask application to top level, moved src to web.

dlahtine committed:
Merge branch 'dev' of https://github.com/zawata/S19-Team2 into flask_server

dlahtine committed:
Created endpoints for sun, moon, spacecraft position and orientation.

Created endpoints for accessing and replacing SPK file.

implemented functionality for serving webpages.

dlahtine committed:
Merge branches 'dev' and 'flask_server' of https://github.com/zawata/S19-Team2 into flask_server

dlahtine committed:
Moved flask application to top level, moved src to web.

dlahtine committed:
Task 34: Added file upload functionality, loads spyce (when built beforehand) removed extraneous functions, created file upload endpoint

dlahtine committed:
Merged dev into flask_server

dlahtine committed:
to be under web instead of src

dlahtine committed:
added /data/trajectory

dlahtine committed:
Rearranged items

dlahtine committed:
Changed "spk" to "bsp", the actual format we'll be using

trevorforrey reviewed (changes requested):
It looks like this PR includes all the spyce build files. Would you be able to remove them from the PR?

trevorforrey committed:
Updated run dev and build commands for package.json

trevorforrey committed:
Merge branch 'dev' into flask_server

trevorforrey committed:
Moving images and lensflare into web folder

trevorforrey commented:
I was able to get it running when removing the sypce specific code. Host the page great, I just made some updates in the package.json and got it updated to merge!

zawata committed:
Make a couple changes to the File Handling Behavior

Redirect root to index.html
Fix loading PNG files
Make Static Serving less strict
remove needless endpoint(now covered by the static file endpoint)

zawata reviewed (approved):
I made a couple changes to the way the server handled files but nothing severe. 
Works amazingly for me. 

Approved

This should be merged so we can start work on a docker image.
